libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -753,17 +763,35 @@ def get_num_tokens_from_messages(self, messages: List[BaseMessage]) -> int:
             raise NotImplementedError(
                 f"get_num_tokens_from_messages() is not presently implemented "
                 f"for model {model}. See "
-                "https://platform.openai.com/docs/guides/text-generation/managing-tokens"
+                "https://platform.openai.com/docs/guides/text-generation/managing-tokens"  # noqa: E501
                 " for information on how messages are converted to tokens."
             )
         num_tokens = 0
         messages_dict = [_convert_message_to_dict(m) for m in messages]
         for message in messages_dict:
             num_tokens += tokens_per_message
             for key, value in message.items():
-                # Cast str(value) in case the message value is not a string
-                # This occurs with function messages
-                num_tokens += len(encoding.encode(str(value)))
+                if isinstance(value, list):
+                    for val in value:
+                        if val["type"] == "text" or isinstance(val, str):
```

Comment:
```suggestion
                        if isinstance(val, str) or val["type"] == "text":
```
Would this raise an error as-is if `val` is a str?

Reply:
lol yea good catch. i think technically by the time we're here it can't be a string anyways, but seems worth doing this for future proofing